src/Components/Editor.tsx

```diff
@@ -458,7 +466,7 @@ export default function Editor({
   // ===========================================================================
   // Buttons
   // ===========================================================================
-  const formatCodeButton = (
+  const formatCodeButton = appEngine === 'python' ? (
```

Comment:
I agree, we should definitely look into integrating something like styler. For the moment the idea was mostly just so that a broken format button does not appear when using the R engine.

I think it makes sense to experiment with this in a later PR, as this one is already pretty large.